python/lsst/ts/ess/common/mock_read_loop_data_client.py

```diff
@@ -0,0 +1,106 @@
+# This file is part of ts_ess_common.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+import asyncio
+import logging
+import types
+from typing import TYPE_CHECKING, Any
+
+import yaml
+
+from .base_read_loop_data_client import BaseReadLoopDataClient
+
+if TYPE_CHECKING:
+    from lsst.ts import salobj
+
+__all__ = ["MockReadLoopDataClient"]
+
+
+class MockReadLoopDataClient(BaseReadLoopDataClient):
+    """Trivial concrete subclass of BaseReadLoopDataClient."""
+
+    def __init__(
+        self,
+        config: types.SimpleNamespace,
+        topics: salobj.Controller | types.SimpleNamespace,
+        log: logging.Logger,
+        simulation_mode: int = 0,
+    ) -> None:
+        super().__init__(
+            config=config, topics=topics, log=log, simulation_mode=simulation_mode
+        )
+
+        # Keep track of how often the read_data coro is called to mock reading
+        # data.
+        self.num_read_data = 0
+
+        # Future to monitor for results.
+        self.data_read_future: asyncio.Future | None = None
+
+        # This is set to a low number so the unit tests won't take much time.
+        self.max_read_timouts = 2
+
+        # This is set to a low number so the unit tests won't take much time.
+        self.read_sleep_time = 0.2
+
+        # This is set to a low number so the unit tests won't take much time.
+        # For mocking timeouts, set this to a sufficiently high number.
+        self.timeout_interval = 0.1
```

Comment:
Reading the code and the unit test as far as I can see this merely acts as a boolean flag: time out if timeout_interval > read_sleep_time else don't time out. In which case I think it would be much clearer to make this truly a flag, e.g.:
```
# Set this to True in order to make read_data time out.
self.make_read_data_time_out = False
```

Reply:
Much clearer indeed.